crates/ra_project_model/src/lib.rs

```diff
@@ -58,22 +58,16 @@ pub enum ProjectWorkspace {
 #[derive(Clone)]
 pub struct PackageRoot {
     /// Path to the root folder
-    path: PathBuf,
+    pub path: PathBuf,
     /// Is a member of the current workspace
-    is_member: bool,
+    pub is_member: bool,
```

Comment:
Why `pub` if we don't mutate it after constructor ?

Reply:
Right, unfortunately, I cannot put `readonly` in front of a property. This is `pub`, so that we don't copy the `.path` property here and there but just move it out of the temporary `PackageRoot`. Though that the special deconstructor fn for that would seem out-of-place...

Reply:
Thats why we use accessor method in the first place.

Reply:
And there is no significant different between `a.foo` and `a.foo()` if `foo()` return a ref.

Reply:
The difference is that you cannot move out values from the ref. E.g.:
```rust
fn consume(p: PathBuf) {};
consume(a.foo);
consume(a.foo()); // cannot move out of shared reference
```

Reply:
Yes, but we only move out once, and I don't think it is so significant.

Reply:
Fixed https://github.com/rust-analyzer/rust-analyzer/pull/3798/commits/ac699dacd163c7234ff7c431d269e9a76bb83878